bin/run.sh

```diff
@@ -1,57 +1,65 @@
+
 path="/var/www/html/php-phenomenal-portal-app-library"
-markdownFolder="$path/wiki-markdown"
 htmlFolder="$path/wiki-html"
-gitList="$path/conf/gitList.txt"
-extension=".html"
-branchConfigFile="$path/conf/branch.config"
-
-if [ -n "${GIT_BRANCH}" ]; then
-  echo "Setting branch on $branchConfigFile file"
-  echo "export BRANCH=$GIT_BRANCH" > $branchConfigFile
-fi
+markdownFolder="$path/wiki-markdown"
+remoteGitList="https://raw.githubusercontent.com/phnmnl/portal-settings/master/app-library/gitList.txt"
+gitBranch="master"
 
-# Setting the variable and then sourcing it is because cron doesn't get the same env as when this is run
-# directly. First part is triggered when running the first time, sourcing is used in that case and when running 
-# through cron.
+# set directories to host new files
+timestamp="$(date +%s)"
+newHtmlFolder="${htmlFolder}-${timestamp}"
+newMarkdownFolder="${markdownFolder}-${timestamp}"
 
-source $branchConfigFile
-if [ -z ${BRANCH+x} ]; then
-  echo "BRANCH var is unset, setting to default master"
-  BRANCH=master
+# read the old html link
+if [[ -L ${htmlFolder} ]]; then
+    oldHtmlFolder=$(readlink -f ${htmlFolder})
 fi
 
-echo "Using $BRANCH branch for portal settings and README files"
-
-wget -O $gitList https://raw.githubusercontent.com/phnmnl/portal-settings/$BRANCH/app-library/gitList.txt
-
-mkdir -p $markdownFolder
-mkdir -p $htmlFolder
+# read the old markdown link
+if [[ -L ${markdownFolder} ]]; then
+    oldMarkdownFolder=$(readlink -f ${markdownFolder})
+fi
 
-cd $markdownFolder && rm -rf *
+# print path info
+echo "Script path: ${current_path}"
+echo "Target base path: ${path}"
+echo "Html Folder [New]: ${newHtmlFolder}"
+echo "Html Folder [Old]: ${oldHtmlFolder}"
+echo "Html Folder [Link]: ${htmlFolder}"
+echo "Markdown Folder [New]: ${newMarkdownFolder}"
+echo "Markdown Folder [Old]: ${oldMarkdownFolder}"
+echo "Markdown Folder [Link]: ${markdownFolder}"
 
-echo $gitList
+# start conversion
+${converter} \
+    --force-cleanup \
+    --html "${newHtmlFolder}" \
+    --md "${newMarkdownFolder}" \
+    --git-branch "${gitBranch}" \
+    "${remoteGitList}"
 
-while IFS= read line
-do
-    git clone -b $BRANCH --depth 1 "$line"
-done <"$gitList"
 
-PATH=/usr/local/bin/:$PATH
+# check whether there exists the new folder
+if [[ -d ${newHtmlFolder} && -d ${newMarkdownFolder} ]]; then
 
-for dir in `ls ./`;
-do
-    mkdir -p "$htmlFolder/$dir"
-    for file in `ls ./$dir`;
-    do
-      filename="${file%.*}"
-      if [[ $filename =~ (README.*|\.md) ]]; then
-	 echo "Transforming $dir/$file"
-         markdown2 --extras fenced-code-blocks "$dir/$file" > "$htmlFolder/$dir/$filename"
-         cp "$htmlFolder/$dir/$filename" "$htmlFolder/$dir/$filename$extension"
-      else
-	 cp -r "$dir/$file" "$htmlFolder/$dir/$file"
-      fi
+    echo -e "\nCreating links to the updated resources..."
+    echo " - Linking new markdown folder ${newMarkdownFolder}"
+    ln -sfn ${newMarkdownFolder} ${markdownFolder}
```

Comment:
quotes here and below

Reply:
Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/80dbfc1679f9100979ee9b7eacc263546852d30f